### PR TITLE
feat: flexible column layout

### PR DESCRIPTION
**Pull request by JKMarkowski**
## Related Issue
Closes SAP/fundamental-styles#1878

## Description
There is added `flexible column layout` component with delta theming and examples of usage.

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [NA] focus state of the element follow design spec
- [NA] active state of the element follow design spec
- [NA] selected state of the element follow design spec
- [NA] selected hover state of the element follow design spec
- [NA] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [NA] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
4. Documentation
- [x] Storybook documentation has been created/updated
- [x] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

File: src/flexible-column-layout.scss [hunk 1]
```diff
@@ -0,0 +1,98 @@
+@import "./new-settings";
+@import "./mixins";
+
+$block: #{$fd-namespace}-flexible-column-layout;
+
+.#{$block} {
+
+  $column-layout-expand-decoration-image: var(--fdFlexibleColumnLayoutToggleImage);
+
+  $toggle-button-height: 2rem;
```
Comment (InnaAtanasova) [inline src/flexible-column-layout.scss:10]: these 2 should have the `fd-` prefix in the naming

File: src/flexible-column-layout.scss [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+  @mixin both-pseudo-selectors() {
+    &::after,
+    &::before {
+      @content;
+    }
+  }
```
Comment (InnaAtanasova) [inline src/flexible-column-layout.scss:17]: maybe move this to the `_mixins.scss` file

File: src/flexible-column-layout.scss [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      @include both-pseudo-selectors() {
+        height: 7rem;
+      }
+    }
+  }
+
+  .#{$block}__button {
+    height: $toggle-button-height;
+    width: 1.5rem;
+    min-width: 1.5rem;
```
Comment (InnaAtanasova) [inline src/flexible-column-layout.scss:60]: The value should be a variable

File: src/flexible-column-layout.scss [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      display: block;
+      width: 0.0625rem;
+      height: 4rem;
+      transition: all 0.1s ease-in;
+
+      @include fd-absolute-horizontal-center();
+    }
+
+    &::before {
+      bottom: calc(50% + #{$toggle-button-height} / 2);
```
Comment (InnaAtanasova) [inline src/flexible-column-layout.scss:41]: This can be a variable and used in bottom and top

File: src/mixins/_mixins.scss [hunk 5]
```diff
@@ -324,6 +324,21 @@
...
+  @content;
+}
+
+@mixin fd-absolute-center {
+  position: absolute;
+  left: 50%;
+  top: 50%;
+  transform: translate(-50%, -50%);
+  @content;
+}
```
Comment (InnaAtanasova) [inline src/mixins/_mixins.scss:340]: I don't think @content is appropriate in these cases

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

**Comment by JKMarkowski** [comment]
@InnaAtanasova Thank You for checking, you comments has been addressed, could you take a look again? :)

**Comment by stefanoScalzo** [comment]
Screen Shot 2020-11-17 at 12 11 42 PM
Screen Shot 2020-11-17 at 12 11 38 PM
The button is going behind

**Review by katekozlowska — APPROVED** [review]
Good job with this component! Only one thing - required value `min-width: 20rem,` makes presented example look broken (f.e. ratio 1:2 in two-columns layout looks incorrect), but it works fine, when I'm testing it separately on different examples.

File: src/mixins/_mixins.scss [hunk 6]
```diff
@@ -351,6 +364,13 @@
   font-size: $fd-size;
 }
 
+@mixin both-pseudo-selectors() {
```
Comment (InnaAtanasova) [inline src/mixins/_mixins.scss:367]: sorry, could you add `fd-` for consistency?

**Review by InnaAtanasova — APPROVED** [review]

**Comment by stefanoScalzo** [comment]
LGTM

**Review by stefanoScalzo — APPROVED** [review]

**Pull request merged**